### PR TITLE
feat(datasource/conan): add sourceUrl when artifactory is used

### PR DESCRIPTION
**Pull request by MartijnLeijssen-TomTom**
## Changes

When using `conan` datasource and using `artifactory` to upload the packages, the sourceUrl is fetched from the conan recipe property `conan.package.url` which is set on artifactory.

## Context

Create a `Release Notes` chapter in the PR's that are being raised by renovate for Conan packages. 
The initial discussion can be found [here](https://github.com/renovatebot/renovate/discussions/22625).

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/conan/index.ts [hunk 1]
```diff
@@ -143,6 +151,39 @@ export class ConanDatasource extends Datasource {
...
             }
           }
+
+          if (isArtifactoryServer(rep)) {
+            const conanApiRegexp = /(?<host>.*)\/artifactory\/api\/conan\/(?<repo>[^\/]+)/
+            const groups = url.match(conanApiRegexp)?.groups
+            if (!groups) {
+              return dep
+            }
+            const versioning = 'semver'
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:161]: is Conan using strict semver? or can there ba other versioning?

Comment (MartijnLeijssen-TomTom) [reply lib/modules/datasource/conan/index.ts:161]: Nice question, this is something I don't really like in this code. Conan doesn't strictly use semver, but it is the [default ](https://docs.conan.io/1/creating_packages/define_abi_compatibility.html#versioning-schema). I wouldn't mind changing this, but I couldn't find a way to get the versioning information from the config object to this datasource without breaking the interface, any suggestions?

File: lib/modules/datasource/conan/index.ts [hunk 2]
```diff
@@ -143,6 +151,43 @@ export class ConanDatasource extends Datasource {
...
+            const version = allVersioning.get(versioning);
+
+            const sortedReleases = dep.releases
+              .filter((release) => version.isVersion(release.version))
+              .sort((a, b) => version.sortVersions(a.version, b.version));
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:166]:
```suggestion
              .filter((release) => semver.isVersion(release.version))
              .sort((a, b) => semver.sortVersions(a.version, b.version));
```

File: lib/modules/datasource/conan/index.ts [hunk 3]
```diff
@@ -130,7 +139,6 @@ export class ConanDatasource extends Datasource {
         if (versions) {
           logger.trace({ lookupUrl }, 'Got conan api result');
           const dep: ReleaseResult = { releases: [] };
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:141]:
```suggestion
           const dep: ReleaseResult = { releases: [] };

```

File: lib/modules/datasource/conan/index.ts [hunk 4]
```diff
@@ -130,7 +139,6 @@ export class ConanDatasource extends Datasource {
...
-
           for (const resultString of Object.values(versions.results ?? {})) {
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:142]:
```suggestion
 
          for (const resultString of Object.values(versions.results ?? {})) {
```

**Review by viceice — DISMISSED** [review]

File: lib/modules/datasource/conan/index.ts [hunk 5]
```diff
@@ -130,7 +139,7 @@ export class ConanDatasource extends Datasource {
         if (versions) {
           logger.trace({ lookupUrl }, 'Got conan api result');
           const dep: ReleaseResult = { releases: [] };
-
+ 
           for (const resultString of Object.values(versions.results ?? {})) {
```
Comment (viceice) [inline lib/modules/datasource/conan/index.ts:143]:
```suggestion

          for (const resultString of Object.values(versions.results ?? {})) {
```

**Review by viceice — APPROVED** [review]

**Pull request merged**